packages/imperative/src/config/src/ProfileInfo.ts

```diff
@@ -316,8 +316,13 @@ export class ProfileInfo {
                     this.getTeamConfig().api.layers.activate(osLoc.user, osLoc.global);
                 }
 
+                const updateVaultOnly = options.setSecure && this.getTeamConfig().api.secure.secureFields().includes(toUpdate.argLoc.jsonLoc);
                 this.getTeamConfig().set(toUpdate.argLoc.jsonLoc, options.value, { secure: options.setSecure });
-                await this.getTeamConfig().save(false);
+                if (!updateVaultOnly) {
+                    await this.getTeamConfig().save(false);
+                } else {
+                    await this.getTeamConfig().api.secure.save(false);
+                }
```

Comment:
This probably sounds picky, but what are your thoughts on inverting the logic for the `if/else` for clarity? 😋 

```suggestion
                if (updateVaultOnly) {
                    await this.getTeamConfig().api.secure.save(false);
                } else {
                    await this.getTeamConfig().save(false);
                }
```

Reply:
I like this idea, but am reluctant to make the change here without making the same change in the V3 branch 😋

Reply:
Ah, fair point - must have missed this in the v3 PR. Disregard this then, thanks for the consideration 😁